### PR TITLE
server: split code and improve ci performance

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #44940

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: server/internal/util/util.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+package util
```
Comment (Defined2014) [inline server/internal/util/util.go:36]: Many functions in this package could be replaced by `util/codec` and `binary` packages. We could do it later.

**Review by Defined2014 — APPROVED** [review]

File: server/internal/util/util.go (hunk 1 above)
Comment (hawkingrei) [reply server/internal/util/util.go:36]: Thank you, It is a good suggestion.

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**